src/traversal/mod.rs

```diff
@@ -166,6 +166,50 @@ pub fn bfs_successors(
     }
 }
 
+/// Return predecessors in a breadth-first-search from a source node.
+///
+/// The return format is ``[(Parent Node, [Children Nodes])]`` in a bfs order
+/// from the source node provided.
+///
+/// :param PyDiGraph graph: The DAG to get the bfs_predecessors from
+/// :param int node: The index of the dag node to get the bfs predecessors for
+///
+/// :returns: A list of nodes's data and their children in bfs order. The
+///     BFSPredecessors class that is returned is a custom container class that
+///     implements the sequence protocol. This can be used as a python list
+///     with index based access.
+/// :rtype: BFSPredecessors
+#[pyfunction]
+#[pyo3(text_signature = "(graph, node, /)")]
+pub fn bfs_predecessors(
+    py: Python,
+    graph: &digraph::PyDiGraph,
+    node: usize,
+) -> iterators::BFSPredecessors {
+    let index = NodeIndex::new(node);
+    let reverse_graph = Reversed(&graph.graph);
+    let mut bfs = Bfs::new(reverse_graph, index);
+    let mut out_list: Vec<(PyObject, Vec<PyObject>)> = Vec::with_capacity(graph.node_count());
+    while let Some(nx) = bfs.next(reverse_graph) {
+        let children = graph
+            .graph
+            .neighbors_directed(nx, petgraph::Direction::Incoming);
+        let mut predecessors: Vec<PyObject> = Vec::new();
+        for pred in children {
+            predecessors.push(graph.graph.node_weight(pred).unwrap().clone_ref(py));
+        }
```

Comment:
I realize this was just copied and pasted from the `bfs_successors` function, but we can improve the efficiency here by using the iterator to build the vector all at once. 

```suggestion
        let predecessors: Vec<PyObject> = graph
            .graph
            .neighbors_directed(nx, petgraph::Direction::Incoming)
            .map(|pred| graph.graph.node_weight(pred).unwrap().clone_ref(py))
            .collect();
```

Tthe `bfs_successors` function was one of the first functions in retworkx so I wasn't super experienced with things when I wrote it. Could you also push a follow up to make this improvement to `bfs_successors` too.

Reply:
Yes, I can update it there too. Thanks!